### PR TITLE
FIX: Plugin JS failing to load would break admin interface

### PR DESCRIPTION
**Pull request by martin-brennan**
If a plugin's JS fails to load for some reason, most commonly
ad blockers, the entire admin interface would break. This is because
we are adding links to the admin routes for plugins that define
them in the sidebar.

We have a fix for this already in the plugin list which shows a warning
to the admin. This fix just prevents the broken link from rendering
in the sidebar if the route is not valid.

### CODE REVIEW

File: app/assets/javascripts/discourse/app/lib/sidebar/admin-sidebar.js [hunk 1]
```diff
@@ -269,9 +271,26 @@ export function addAdminSidebarSectionLink(sectionName, link) {
...
+      }
+
+      // Check if the admin route is valid, if it is not the whole admin
+      // interface can break because of this. This can be the case for things
+      // like ad blockers stopping plugin JS from loading.
+      if (adminRouteValid(router, plugin.admin_route)) {
+        return true;
+      } else {
+        warn(
+          `[AdminSidebar] Could not find admin route for ${plugin.name}, route was ${plugin.admin_route.full_location}`,
```
Comment (Drenmi) [inline app/assets/javascripts/discourse/app/lib/sidebar/admin-sidebar.js:288]: Should we mention ad blocking as a common cause here?

File: lib/plugin/instance.rb [hunk 2]
```diff
@@ -117,6 +117,20 @@ def add_admin_route(label, location, opts = {})
...
+    return unless route
+
+    ret = route.slice(:location, :label)
+    if route[:use_new_show_route]
+      ret[:full_location] = "adminPlugins.show"
+      ret[:use_new_show_route] = true
+    else
+      ret[:full_location] = "adminPlugins.#{ret[:location]}"
+    end
+    ret
```
Comment (Drenmi) [inline lib/plugin/instance.rb:131]: I think right now the `use_new_show_route` key is entirely omitted if the value is `false`. I think it's better to still include it and set the value to `false`. WDYT?

```suggestion
    route.slice(:location, :label, :use_new_show_route).tap do |admin_route|
      path = admin_route[:use_new_show_route] ? "show" : admin_route[:location]

      admin_route[:full_location] = "admin_plugins.#{path}"
    end
```

**Review by Drenmi — APPROVED** [review]
🪨

File: app/assets/javascripts/discourse/app/lib/sidebar/admin-sidebar.js (hunk 1 above)
Comment (martin-brennan) [reply app/assets/javascripts/discourse/app/lib/sidebar/admin-sidebar.js:288]: Yep good idea, will add

File: lib/plugin/instance.rb (hunk 2 above)
Comment (martin-brennan) [reply lib/plugin/instance.rb:131]: Looks nice, will use this :+1:

File: app/assets/javascripts/discourse/app/lib/sidebar/admin-sidebar.js [hunk 3]
```diff
@@ -269,9 +271,26 @@ export function addAdminSidebarSectionLink(sectionName, link) {
...
+
+      // Check if the admin route is valid, if it is not the whole admin
+      // interface can break because of this. This can be the case for things
+      // like ad blockers stopping plugin JS from loading.
+      if (adminRouteValid(router, plugin.admin_route)) {
+        return true;
+      } else {
+        warn(
+          `[AdminSidebar] Could not find admin route for ${plugin.name}, route was ${plugin.admin_route.full_location}. This could be caused by an ad blocker.`,
+          { id: "admin-sidebar:pluginAdminRouteLinks" }
```
Comment (tyb-talks) [inline app/assets/javascripts/discourse/app/lib/sidebar/admin-sidebar.js:289]: Was looking through prior usage of this - should we use the `discourse` prefix for the id here to make it clearer it's from discourse? And would it be better to standardize on kebab case?

Comment (martin-brennan) [reply app/assets/javascripts/discourse/app/lib/sidebar/admin-sidebar.js:289]: Yah thats fair, will fix that up :+1:

**Review by tyb-talks — APPROVED** [review]

**Pull request merged**